ts/private/ts_project.bzl

```diff
@@ -158,17 +162,23 @@ This might be because
 This is an error because Bazel does not run actions unless their outputs are needed for the requested targets to build.
 """)
 
-    typings_outputs = typings_outs + typing_maps_outs + typings_srcs
+    declarations = typings_outs + typing_maps_outs + typings_srcs
+
+    # Default outputs (DefaultInfo files) is what you see on the command-line for a built
+    # library, and determines what files are used by a simple non-provider-aware downstream
+    # library. Only the JavaScript outputs are intended for use in non-TS-aware dependents.
     if ctx.attr.transpile:
-        default_outputs_depset = depset(runtime_outputs) if len(runtime_outputs) else depset(typings_outputs)
+        # Special case case where there are no source outputs and we don't have a custom
+        # transpiler so we add declarations to the default outputs
+        default_outputs = sources[:] if len(sources) else declarations[:]
     else:
         # We must avoid tsc writing any JS files in this case, as tsc was only run for typings, and some other
         # action will try to write the JS files. We must avoid collisions where two actions write the same file.
         arguments.add("--emitDeclarationOnly")
 
         # We don't produce any DefaultInfo outputs in this case, because we avoid running the tsc action
-        # unless the DeclarationInfo is requested.
-        default_outputs_depset = depset([])
+        # unless the declarations are requested.
```

Comment:
did you verify if this continues to work? If some downstream action asks for the JsInfo but doesn't need declarations, are we now going to trigger typechecks where we didn't previously?

(that may be a critical reason to keep the DeclarationInfo separate from JsInfo)

Reply:
I'll double check but yes it should work. A declaration file is only created if it ends up as an input to an action. Having File objects passed around in `declarations` and `transitive_declarations` in JsInfo doesn't trigger the actions unless they are passed as inputs to actions that need to run or as runfiles to binary targets that are run.

Reply:
I verified this locally and pretty sure it does work correctly, since `include_declarations` is False by default